### PR TITLE
[ODS-4579] Update packages in the ODS/API Solution

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/StepsHelper.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 namespace EdFi.Ods.WebApi.CompositeSpecFlowTests
 {
-    public static class StepsHelper
+    public static class StepsHelper
```
Comment (gmcelhanon) [inline tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/StepsHelper.cs:21]: Remove spurious whitespace change.

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@
 using System.Threading.Tasks;
 using ApprovalTests;
 using ApprovalTests.Reporters;
+using Castle.Core.Internal;
```
Comment (gmcelhanon) [inline tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs:15]: Should not be using extension methods from `Castle.Core.Internal`.

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs [hunk 3]
```diff
@@ -560,7 +561,7 @@ private async Task SetIdAsync(string query)
             var id = await StepsHelper.GetAsync<Guid>(
                 _connectionStringProvider.Value.GetConnectionString(), query, _cancellationToken.Value);
 
-            id.ShouldNotBeNull();
+            id.ToString().IsNullOrEmpty().ShouldBeFalse();
```
Comment (gmcelhanon) [inline tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs:564]: What was the reason for this change? It looks like you ended up using an extension method from `Castle.Core.Internal` here -- which is undesirable.

**Review by gmcelhanon — CHANGES_REQUESTED** [review]

File: Application/EdFi.Common/EdFi.Common.csproj [hunk 4]
```diff
@@ -13,9 +13,9 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.3" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="5.0.0" />
```
Comment (gmcelhanon) [inline Application/EdFi.Common/EdFi.Common.csproj:16]: @jamessmckay In the JIRA ticket you wrote there should be no major version upgrades. Does this constitute a major version upgrade?

File: Application/EdFi.Ods.Sandbox/EdFi.Ods.Sandbox.csproj [hunk 5]
```diff
@@ -14,16 +14,16 @@
...
+    <PackageReference Include="Npgsql" Version="4.1.5" />
     <PackageReference Include="log4net" Version="2.0.12" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="3.1.3" />
-    <PackageReference Include="Npgsql" Version="4.1.3.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="5.0.0" />
+    <PackageReference Include="Npgsql" Version="4.1.5" />
     <PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
     <PackageReference Include="System.Buffers" Version="4.5.1" />
-    <PackageReference Include="System.ComponentModel.Annotations" Version="4.7.0" />
+    <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Sandbox/EdFi.Ods.Sandbox.csproj:26]: @jamessmckay In the JIRA ticket you wrote there should be no major version upgrades. Does this constitute a major version upgrade?

File: Application/EdFi.Ods.Security/EdFi.Ods.Security.csproj [hunk 6]
```diff
@@ -14,21 +14,21 @@
...
-    <PackageReference Include="System.Configuration.ConfigurationManager" Version="4.7.0" />
+    <PackageReference Include="System.Configuration.ConfigurationManager" Version="5.0.0" />
     <PackageReference Include="System.Data.DataSetExtensions" Version="4.5.0" />
-    <PackageReference Include="System.ComponentModel.Annotations" Version="4.7.0" />
+    <PackageReference Include="System.ComponentModel.Annotations" Version="5.0.0" />
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="6.0.0" />
+    <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="7.1.0" />
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Security/EdFi.Ods.Security.csproj:31]: @jamessmckay In the JIRA ticket you wrote there should be no major version upgrades. Does this constitute a major version upgrade?

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/StepsHelper.cs (hunk 1 above)
Comment (semalaiappan) [reply tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/StepsHelper.cs]: removed

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs (hunk 2 above)
Comment (semalaiappan) [reply tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs]: removed

File: tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs (hunk 3 above)
Comment (semalaiappan) [reply tests/EdFi.Ods.WebApi.CompositeSpecFlowTests/Steps.cs]: fixed it . Thanks

File: Application/EdFi.Common/EdFi.Common.csproj (hunk 4 above)
Comment (semalaiappan) [reply Application/EdFi.Common/EdFi.Common.csproj]: I limited to  minor upgrades . thanks

File: Application/EdFi.Ods.Sandbox/EdFi.Ods.Sandbox.csproj (hunk 5 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Sandbox/EdFi.Ods.Sandbox.csproj]: I limited to  minor upgrades . thanks

File: Application/EdFi.Ods.Security/EdFi.Ods.Security.csproj (hunk 6 above)
Comment (semalaiappan) [reply Application/EdFi.Ods.Security/EdFi.Ods.Security.csproj]: I limited to  minor upgrades . thanks

**Review by gmcelhanon — APPROVED** [review]

**Pull request merged**